File: Client/src/app/features/vendors/data-access/models/update-vendor-response.ts
import { Address } from "../../../../shared/models/address-model";
import { ContactResponse } from "../../../../shared/models/contacts-response";


export interface UpdateVendorResponse {
  readonly id: string;
  readonly name: string;
  readonly address: Address;
  readonly webpage: string | null;
  readonly contacts: readonly ContactResponse[];
}

File: AssetManagement.Application/Vendors/UpdateVendor/UpdateVendorResponse.cs
using AssetManagement.Application.Common.Models;

namespace AssetManagement.Application.Vendors.UpdateVendor
{
    public sealed record UpdateVendorResponse(
        Guid Id,
        string Name,
        AddressResponse Address,
        string? Webpage,
        IReadOnlyCollection<ContactResponse> Contacts);
}
